modules/icinga2/api.go

```diff
@@ -0,0 +1,119 @@
+package icinga2
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/NETWAYS/support-collector/internal/collection"
+	"io"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type UserAuth struct {
+	Username string
+	Password string
+}
+
+// ApiCred saves the user and password. Provided as arguments
+var ApiCred UserAuth
+
+// ApiEndpoints saves the FQDN or ip address for the endpoints, that will be collected. Provided as arguments.
+var ApiEndpoints []string
+
+// InitApiCollection starts to collect data from the Icinga 2 API for given endpoints
+func InitApiCollection(c *collection.Collection) error {
+	// return if no endpoints are provided
+	if len(ApiEndpoints) <= 0 {
+		return fmt.Errorf("0 API endpoints provided. No data will be collected from remote targets")
+	}
+	c.Log.Info("Start collection of Icinga 2 API endpoints")
+
+	// return if username or password is not provided
+	if ApiCred.Username == "" || ApiCred.Password == "" {
+		return fmt.Errorf("API Endpoints provided but username and/or password are missing")
+	}
+
+	for _, endpoint := range ApiEndpoints {
+		// check if endpoint is reachable
+		err := endpointIsReachable(endpoint)
+		if err != nil {
+			c.Log.Warn(err)
+			continue
+		}
+		c.Log.Debugf("Endpoint '%s' is reachable", endpoint)
+
+		// collect /v1/status from endpoint
+		err = collectStatus(endpoint, c)
+		if err != nil {
+			c.Log.Warn(err)
+		}
+	}
+
+	return nil
+}
+
+// endpointIsReachable checks if the given endpoint is reachable within 5 sec
+func endpointIsReachable(endpoint string) error {
+	timeout := 5 * time.Second
+
+	// try to dial tcp connection within 5 seconds
+	conn, err := net.DialTimeout("tcp", endpoint, timeout)
+	if err != nil {
+		return fmt.Errorf("cant connect to endpoint '%s' within 5 seconds: %s", endpoint, err)
+	}
+	defer conn.Close()
+
+	return nil
+}
+
+// collectStatus requests $endpoint$/v1/status with ApiCred and saves the json result to file
+func collectStatus(endpoint string, c *collection.Collection) error {
+	c.Log.Debugf("request data from endpoint '%s/v1/status'", endpoint)
+
+	// allow insecure connections because of Icinga 2 certificates
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+
+	// build request
+	req, err := http.NewRequest("GET", fmt.Sprintf("https://%s/v1/status", endpoint), nil)
```

Comment:
We should use a context here for timeout handling

https://pkg.go.dev/net/http#NewRequestWithContext